### ISSUE TITLE
Overland not updating from early build

### ISSUE THREAD

**Issue by kristenpen** [itchio/itch#604]
Installed overland from earlier itch app release. I cant figure out how to update it, clicking 'launch' starts old version

Also it would be nice if it showed which version was installed, directly in the app

ask leaf for more info

**Comment by fasterthanlime** [comment]
As of v17.1.0 (still building as I'm posting this), here's what the dropdown shows:

<img width="538" alt="screen shot 2016-05-06 at 11 14 43" src="https://cloud.githubusercontent.com/assets/7998310/15069109/caffdbd8-137b-11e6-9c0e-372d4f5a9833.png">

Clicking on the 'Version' entry copies it to your clipboard, so that you can paste it into issues reports.

Clicking on 'Check for update' forces an update check and will let you know (in the status bar) if an update has been found or not. I've also made the update check more robust.

If it still can't update Overland to the latest version:
- Please paste the version number currently installed (get it via the dropdown)
- Please attach the latest log file in `%USERDATA%\itch\logs\`

**Comment by leafo** [comment]
Thanks for the update to the down arrow.

We ended up deleing and reinstalling. The version number was 17.0.0. I don't have access to that computer so I can get you logs.

Do you think this was a specific case? I grantee if someone else gets this bug they're not going to know to dig thorough the down menu at first, and they'll be frustrated/confused. Is it related to having an install from the old version of the app?

**Comment by fasterthanlime** [comment]
> Do you think this was a specific case? I grantee if someone else gets this bug they're not going to know to dig thorough the down menu at first, and they'll be frustrated/confused. Is it related to having an install from the old version of the app?

Here's what I think happened: versions 0.14.x of the app were able to download+install a wharf-generated .zip, but didn't store which `buildId` was installed.

The update logic was:
- If cave has a `buildId`, compare buildIds (wharf update)
- If cave doesn't, compare uploadIds

So it went through the second path, and, seeing as the uploadId stayed the same, it didn't update.

Since v17.1.0, if the API response for the upload that we have locally installed contains a `buildId`, and the cave doesn't have a `buildId` field (ie. it was installed with v0.14.x of the app) then it considers it stale and redownloads the latest .zip, correctly storing the `buildId`.

**TL;DR v17.1.0 fixes this issue without people having to dig through dropdown.**
